### ISSUE TITLE
Create attributes for absolute and relative gap tolerances

### ISSUE THREAD

**Issue by mtanneau** [jump-dev/MathOptInterface.jl#1936]
[EDIT] following the discussion and the distinction between model and optimizer attributes, I have renamed the PR and updated the proposals.

___

MOI currently has a [`RelativeGap`](https://jump.dev/MathOptInterface.jl/stable/reference/models/#MathOptInterface.RelativeGap) attribute for relative optimality gap.

Several solvers also support an absolute gap tolerance, and it is not uncommon to use both relative and absolute gap tolerances.

If the feature gets a green light, I can PR the corresponding changes and docs.

**Comment by odow** [comment]
I think that's reasonable. It's a bit easier to define a fallback for as well.

**Comment by mtanneau** [comment]
Precision: I want to use this not just to query the absolute gap after the optimization is done, but also to set it as tolerance for the solver.

That latter use case may fall under `AbstractOptimizerAttribute` rather than `AbstractModelAttribute`, which is the current super type of `MOI.RelativeGap`

**Comment by odow** [comment]
See my comment here: https://github.com/jump-dev/Gurobi.jl/pull/484#issuecomment-1175579115

These are conflating two separate things. Not sure what the best solution is.

Querying is straight forward, and I'd accept that as a PR under `AbstractModelAttribute`. Setting might need a larger discussion.

**Comment by odow** [comment]
Linking to previous discussions:

 * https://github.com/jump-dev/MathOptInterface.jl/issues/1012
 * https://github.com/jump-dev/MathOptInterface.jl/issues/61

**Comment by mtanneau** [comment]
I see. 
I can propose to introduce `RelativeGapTolerance` and `AbsoluteGapTolerance` as optimizer attributes.
Absolute gap tolerance should be relatively solver-independent :) Relative gap tolerance is solver-dependent, same as `RelativeGap` already is.

I personally use these two only in the context of MIP, although such tolerances may be used in global optimization, or iterative algorithms like Benders decomposition, Frank-Wolfe, column generation, etc.
___

For context, this is more-or-less how we use them internally (though technically we miss-used `RelativeGap`):
```julia
set_optimizer_attribute(model, MOI.RelativeGap(), 1e-3)
set_optimizer_attribute(model, AbsoluteGap(), 10_000)
```

Having this exposed at the MOI level would make our life easier, but we can keep our current workflow.

**Comment by blegat** [comment]
> I can propose to introduce `RelativeGapTolerance` and `AbsoluteGapTolerance` as optimizer attributes.

That sounds good to me

**Comment by odow** [comment]
Closed by #1944
